### ISSUE TITLE
possible bug in example code p. 27

### ISSUE THREAD

**Issue by sbramson** [laboon/ebook#133]
as currently written, the array blub2 is longer than just the "CHIRP!" (carefully and correctly placed at its end). It also holds a bunch of characters before that, and these are never initialized. Depending on the behavior of the String(char[] arr) constructor, those <NUL>'s might still be in the concatenated result: boxer99.

**Comment by timparenti** [comment]
Although the characters in `blub2` are not explicitly initialized, all six of its characters are immediately defined in the lines that follow its declaration.  There are no other elements in the character array.
```java
       blub2[0] = (char) 0x43;
       blub2[1] = (char) 0110;
       blub2[2] = (char) 73;
       blub2[3] = (char) (0123 - 1);
       blub2[4] = (char) (40 * 2);
       blub2[5] = '!';
```

Since Java tracks array bounds, there shouldn't be a need for an explicit string termination character.

**Comment by laboon** [comment]
@timparenti Thanks for chiming in and I'm glad to know that you're still watching this repo! =)

He's actually correct, but in response to a previous commit.  I told him to file an issue, but I then had some free time after lunch and fixed it myself here: https://github.com/laboon/ebook/commit/3be7be36c74495aa7efd5e32394e5276697f4cf7  So we had a race condition regarding filing the issue and fixing it. 

Previously, I generated a larger char array, then concatenated.  For some reason, I had thought that the null chars at the beginning of the array would be eliminated when I created a new String(), but they did not.  However, I tested this just using System.out.println, which does _not_ print out nulls.  So the Strings created in the two `chirpify()` methods _were_ different, but if you just printed them out, you didn't _see_ a difference.  But examining the actual Strings showed that the nulls were in fact there, meaning that the two methods were not exactly the same, as the book said.
